cmd/git-sync/main.go

```diff
@@ -621,6 +630,22 @@ func main() {
 		}
 
 		if initialSync {
+			// Wait for hooks to complete at least once, if not nil, before checking whether to stop program
+			// Assumes that if hook channels are not nil, they will have at least one value before getting closed
+			if exechookChannel != nil {
```

Comment:
I think we should only do this in case of `--one-time` - is there a good reason to check it otherwise?

This also would let us exit with a non-zero code if one or more hooks failed.  Accumulate a bool "hooksOK" - set it to false in case of error and `os.Exit(1)` at the end?

Reply:
For the first question, the statement I'm trying to guarantee is the following: "Assuming git-sync thread has not terminated with an error code, exechook and webhook, if set,  will always run (not necessarily successfully) at least once." If the user git-syncs a hash for whatever reason, the proposed change would break this invariant we wish to establish. However, since tags are not considered commits in the source code, this won't affect my use case and I would be fine with doing this change if you would like.

As for the boolean thing...let me think about it. [TBT]

Reply:
Yeah, I think we can't really make aany guarantees if a sync doesn't succeed, and if it did succeed, `--one-time` is the only really synchronous thing we support.

Reply:
Sounds good, so in the case of termination due to pulling a hash/specific-commit, we will allow for the hook race condition described in this issue to persist. 

In that case, will move checks to '--one-time' block and add the status-code change.